# PREREQUISITE PREPARATION
## Table of Contents
  - [Table of Contents](#table-of-contents)
  - [Enviroments](#Setup-the-environment)
    - [1. Python ](#1-python)
    - [2. Conda](#2-conda)
    - [3. R-base](#3-r-base)
  - [Tools](#tools)
    - [1. GATK ](#1-GATK)
    - [2. Fastqc ](#2-fastqc)
    - [3. Multiqc](#3-multiqc)
    - [4. Trimmomatic](#4-trimmomatic)
    - [5. BWA ](#5-bwa)
    - [6. Samtools](#6-samtools)
    - [7. Picard](#7-picard)
- [Reference](#reference)
    - [1. Human genome assembly GRCh38 ](#1-Human-genome-assembly-GRCh38)
    - [2. Adapter](#2-adapter)
    - [3. Clinvar](#3-clinvar)

## Setup the environments
### 1. Python
Please download and install **python** in Linux from [here](https://discuss.python.org/t/install-python-3-11-9-on-ubuntu/51093) 

**Quick Guide**
```bash
sudo apt install
sudo apt upgrade
sudo apt install python3
python3 --version # Check version installed
```
### 2.Conda
Please download and install **conda** in Linux from [here](https://github.com/conda-forge/miniforge) 

**Quick Guide**
```bash
wget https://github.com/conda-forge/miniforge/releases/latest/download/Miniforge3-Linux-x86_64.sh
bash bash Miniforge3-Linux-x86_64.sh
```
### 3. R-base
Please download and install **R-base** in Linux from [here](https://cran.r-project.org/bin/linux/ubuntu/fullREADME.html) 

**Quick Guide**
```bash
sudo apt update
sudo apt install r-base
R # open R
```
```r
q() # exit R
```

## Tools
### 1. GATK
The Genome Analysis Toolkit (GATK) is a powerful toolkit developed by the Broad Institute for variant discovery and genotyping. It provides robust tools for processing high-throughput sequencing data, including base quality score recalibration, variant calling, and variant filtering.

You can find more [here](https://gatk.broadinstitute.org/hc/en-us/articles/360035889851--How-to-Install-and-use-Conda-for-GATK4), and install using conda [here](https://anaconda.org/bioconda/gatk) or install and built GATK using java, find [here](https://gatk.broadinstitute.org/hc/en-us/articles/360036194592-Getting-started-with-GATK4)

```bash
conda install bioconda::gatk
source activate gatk
```

### 2. fastqc
FastQC is a widely-used quality control tool for high-throughput sequencing data. It generates detailed reports on sequence quality, GC content, adapter contamination, and other metrics, helping researchers assess the integrity of their sequencing data.

You can find more [here](https://github.com/s-andrews/FastQC), and install using conda [here](https://anaconda.org/bioconda/fastqc)

```bash
conda install bioconda::fastqc
```
### 3. multiqc
MultiQC is a versatile tool that aggregates and visualizes reports from multiple bioinformatics tools, including FastQC, alignment tools, and variant callers. It simplifies data interpretation by compiling all results into a single, easy-to-read report.

You can find more [here](https://github.com/MultiQC/MultiQC), and install using conda [here](https://anaconda.org/bioconda/multiqc)

```bash
conda install bioconda::multiqc
```
### 4. Trimmomatic
Trimmomatic is a flexible tool for trimming and preprocessing Illumina sequencing reads. It removes low-quality bases, adapters, and other artifacts, ensuring clean and high-quality data for downstream analysis.

You can find more [here](https://github.com/usadellab/Trimmomatic), and install using conda [here](https://anaconda.org/bioconda/trimmomatic)

```bash
conda install bioconda::trimmomatic
```
### 5. BWA
Burrows-Wheeler Aligner (BWA) is a fast and efficient tool for aligning short sequencing reads to a reference genome. Its algorithms, such as BWA-MEM, are widely used for DNA and RNA sequencing data alignment.

You can find more [here](https://github.com/lh3/bwa), and install using conda [here](https://anaconda.org/bioconda/bwa)

```bash
conda install bioconda::bwa
```

### 6. Samtools
Samtools is a suite of tools for handling SAM/BAM/CRAM files, which are common formats for aligned sequencing reads. It supports sorting, indexing, filtering, and variant calling, making it an essential tool in bioinformatics workflows.

You can find more [here](https://github.com/samtools/samtools), and install using conda [here](https://anaconda.org/bioconda/samtools)

```bash
conda install bioconda::samtools
```

### 7. Picard
Picard is a collection of command-line tools for manipulating high-throughput sequencing data and BAM files. It includes utilities for marking duplicates, collecting alignment metrics, and validating file formats.

You can find more [here](https://github.com/broadinstitute/picard), and install using conda [here](https://anaconda.org/bioconda/picard)

```bash
conda install bioconda::picard
```
## References
### 1. Human genome assembly GRCh38
GRCh38 (hg38): The latest and most accurate human reference genome. You can find [here](https://hgdownload.soe.ucsc.edu/goldenPath/hg38/bigZips/)

```bash
wget https://hgdownload.soe.ucsc.edu/goldenPath/hg38/bigZips/hg38.fa.gz
gunzip hg38.fa.gz
bwa index hg38.fa
samtools faidx hg38.fa
picard.jar CreateSequenceDictionary \
  R=hg38.fa \
  O=hg38.dict
```
### 2. Adapter
An adapter file in trimming is used to remove unwanted adapter sequences from the raw sequencing reads. During sequencing, short adapter sequences are often ligated to the ends of DNA fragments to facilitate their amplification and sequencing. These adapters can interfere with downstream analysis, so they need to be removed.

You can find more [here](https://github.com/usadellab/Trimmomatic)
```bash
sudo apt update
sudo apt install git

git clone https://github.com/usadellab/Trimmomatic
```

### 3. Clinvar
The ClinVar database is a public archive of reports on the relationships among human variations and phenotypes, with supporting evidence. It is a valuable resource for understanding the clinical significance of genetic variants.
You can find it [here](https://ftp.ncbi.nlm.nih.gov/pub/clinvar/vcf_GRCh38/)
```bash
wget https://ftp.ncbi.nlm.nih.gov/pub/clinvar/vcf_GRCh38/clinvar.vcf.gz
wget https://ftp.ncbi.nlm.nih.gov/pub/clinvar/vcf_GRCh38/clinvar.vcf.gz.tbi
```
