# PREREQUISITE PREPARATION
## Table of Contents
  - [Table of Contents](#table-of-contents)
  - [Enviroments](#set-up-the-environment)
    - [1. python ](#1-python)
    - [2. conda](#2-conda)
    - [3. r-base](#3-r-base)
  - [Tools](#tools)
    - [1. fastqc ](#1-fastqc)
    - [2. multiqc](#2-multiqc)
    - [3. Trimmomatic](#3-trimmomatic)
    - [4. bwa ](#4-bwa)
    - [5. samtools](#5-samtools)
    - [6. picard](#6-picard)
- [Reference](#reference)
    - [1. Human genome assembly GRCh38 ](#1-GRCh38)
    - [2. Adapter](#2-adapter)
    - [3. Clinvar](#3-clinvar)

## Setup the environments
### 1. Python
Please download and install **python** in Linux from [here](https://discuss.python.org/t/install-python-3-11-9-on-ubuntu/51093) 

**Quick Guide**
```bash
sudo apt install
sudo apt upgrade
sudo apt install python3
python3 --version # Check version installed
```
### 2.Conda
Please download and install **conda** in Linux from [here](https://github.com/conda-forge/miniforge) 
**Quick Guide**
```bash
wget https://github.com/conda-forge/miniforge/releases/latest/download/Miniforge3-Linux-x86_64.sh
bash bash Miniforge3-Linux-x86_64.sh
```
### 3. R-base
Please download and install **conda** in Linux from [here](https://cran.r-project.org/bin/linux/ubuntu/fullREADME.html) 
**Quick Guide**
```bash
sudo apt update
sudo apt install r-base
R # open R
```
```r
q() # exit R
```

## Tools
### 1. fastqc
You can find more [here] (https://github.com/s-andrews/FastQC), and install using conda [here] (https://anaconda.org/bioconda/fastqc)

```bash
conda install bioconda::fastqc
```
### 2. multiqc
You can find more [here] (https://github.com/MultiQC/MultiQC), and install using conda [here] (https://anaconda.org/bioconda/multiqc)

```bash
conda install bioconda::multiqc
```
### 3. Trimmomatic
You can find more [here] (https://github.com/usadellab/Trimmomatic), and install using conda [here] (https://anaconda.org/bioconda/trimmomatic)

```bash
conda install bioconda::trimmomatic
```
### 4. BWA
You can find more [here] (https://github.com/lh3/bwa), and install using conda [here] (https://anaconda.org/bioconda/bwa)

```bash
conda install bioconda::bwa
```

### 5. Samtools
You can find more [here] (https://github.com/samtools/samtools), and install using conda [here] (https://anaconda.org/bioconda/samtools)

```bash
conda install bioconda::samtools
```

### Picard
You can find more [here] (https://github.com/broadinstitute/picard), and install using conda [here] (https://anaconda.org/bioconda/picard)

```bash
conda install bioconda::picard
```
## Reference
### 1. Human genome assembly GRCh38
GRCh38 (hg38): The latest and most accurate human reference genome. You can find [here] (https://hgdownload.soe.ucsc.edu/goldenPath/hg38/bigZips/)

```bash
wget https://hgdownload.soe.ucsc.edu/goldenPath/hg38/bigZips/hg38.fa.gz
gunzip hg38.fa.gz
bwa index hg38.fa
samtools faidx hg38.fa
picard.jar CreateSequenceDictionary \
  R=hg38.fa \
  O=hg38.dict
```
### 2. Adapter
An adapter file in trimming is used to remove unwanted adapter sequences from the raw sequencing reads. During sequencing, short adapter sequences are often ligated to the ends of DNA fragments to facilitate their amplification and sequencing. These adapters can interfere with downstream analysis, so they need to be removed.

You can find more [here] (https://github.com/usadellab/Trimmomatic)
```bash
sudo apt update
sudo apt install git

git clone https://github.com/usadellab/Trimmomatic
```

### 3. Clinvar
You can find it [here] (https://ftp.ncbi.nlm.nih.gov/pub/clinvar/vcf_GRCh38/)
```bash
wget https://ftp.ncbi.nlm.nih.gov/pub/clinvar/vcf_GRCh38/clinvar.vcf.gz
wget https://ftp.ncbi.nlm.nih.gov/pub/clinvar/vcf_GRCh38/clinvar.vcf.gz.tbi
```